src/transformers/training_tf_args.py

```diff
@@ -0,0 +1,50 @@
+import logging
+from dataclasses import dataclass, field
+from typing import Tuple
+
+from .file_utils import cached_property, is_tf_available, tf_required
+from .training_args import TrainingArguments
+
+
+logger = logging.getLogger(__name__)
+
+if is_tf_available():
+    import tensorflow as tf
+
+
+@dataclass
+class TFTrainingArguments(TrainingArguments):
+    tpu: bool = field(default=False, metadata={"help": "Run the training over TPUs"})
+    optimizer_name: str = field(default="adam", metadata={"help": "Name of a Tensorflow optimizer"})
+    mode: str = field(
+        default="sequence-classification",
+        metadata={"help": 'Type of task, one of "sequence-classification", "token-classification" '},
+    )
+    loss_name: str = field(default="SparseCategoricalCrossentropy", metadata={"help": "Name of a Tensorflow loss"})
+    eval_steps: int = field(default=1000, metadata={"help": "Run an eval every X steps."})
```

Comment:
in PyTorch we basically use `logging_steps` for this in conjunction with `--evaluate_during_training`

I think here we could do the same, no? (Also because I think there's an issue which I've mentioned in the training loop above)

Reply:
It is something I'm not really fan on in the trainer, because I like to have my loss print every "X" steps but I don't want the model to be evaluated as many time as there is a log, that's why I have created a new parameter, to separate the log itself and the evaluation.

Reply:
Ok, I kinda agree, I can change it in the Trainer too then. I wasn't a fan of this either.

Only thing is that it's breaking the compatibility with the old example scripts a little (not a huge deal)

Reply:
Sounds good to me as well (and I don't think we should really guarantee any backward compatibility for the CLI arguments of the example scripts).